protocol/x/affiliates/keeper/msg_server.go

```diff
@@ -18,6 +20,21 @@ func (k msgServer) RegisterAffiliate(ctx context.Context,
 
 func (k msgServer) UpdateAffiliateTiers(ctx context.Context,
 	msg *types.MsgUpdateAffiliateTiers) (*types.MsgUpdateAffiliateTiersResponse, error) {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	unconditionalRevShareConfig, err := k.revShareKeeper.GetUnconditionalRevShareConfigParams(sdkCtx)
+	if err != nil {
+		return nil, err
+	}
+	marketMapperRevShareParams := k.revShareKeeper.GetMarketMapperRevenueShareParams(sdkCtx)
+
+	if !k.revShareKeeper.ValidateRevShareSafety(*msg.Tiers, unconditionalRevShareConfig, marketMapperRevShareParams) {
+		return nil, errorsmod.Wrapf(
+			types.ErrRevShareSafetyViolation,
+			"rev share safety violation",
+		)
+	}
```

Comment:
Are we hooking up this to the actual logic in the other PR (checking authority, calling keeper)?

Reply:
Yes